File: 509-fibonacci-number/509-fibonacci-number.java
class Solution {
    public int fib(int n) {
        if(n == 0)
            return 0;
        int[] memo = new int[]{0, 1};
        for(int i = 2; i < n + 1; i++)
            memo = new int[]{memo[1], memo[1] + memo[0]};

        return memo[1];
    }
}

File: 509-fibonacci-number/509-fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        if n == 0:
            return 0
        memo = (0, 1)
        for _ in range(2, n + 1):
            memo = (memo[-1], memo[-1] + memo[-2])

        return memo[-1]